src/sonic_ax_impl/mibs/ietf/rfc1213.py

```diff
@@ -196,34 +204,54 @@ def reinit_data(self):
 
         self.vlan_name_map, \
         self.vlan_oid_sai_map, \
-        self.vlan_oid_name_map = Namespace.init_namespace_sync_d_vlan_tables(self.db_conn)
-
-        self.rif_port_map, \
-        self.port_rif_map = Namespace.init_namespace_sync_d_rif_tables(self.db_conn)
+        self.vlan_oid_name_map, \
+        self.vlan_oid_namespace = Namespace.init_namespace_sync_d_vlan_tables(self.db_conn)
+        
+        for db_index in Namespace.get_non_host_db_indexes(self.db_conn):
+            self.rif_port_map[db_index], \
+            self.port_rif_map[db_index] = mibs.init_sync_d_rif_tables(self.db_conn[db_index])
 
     def update_data(self):
         """
         Update redis (caches config)
         Pulls the table references for each interface.
         """
-        self.if_counters = \
-            {sai_id: Namespace.dbs_get_all(self.db_conn, mibs.COUNTERS_DB, mibs.counter_table(sai_id), blocking=True)
-            for sai_id in self.if_id_map}
-
-        rif_sai_ids = list(self.rif_port_map) + list(self.vlan_name_map)
+        self.lag_name_if_name_map, \
+        self.if_name_lag_name_map, \
+        self.oid_lag_name_map, \
+        self.lag_sai_map, \
+        self.lag_sai_oid_map = Namespace.init_namespace_sync_d_lag_tables(self.db_conn)
 
-        self.rif_counters = \
-            {sai_id: Namespace.dbs_get_all(self.db_conn, mibs.COUNTERS_DB, mibs.counter_table(sai_id), blocking=True)
-             for sai_id in rif_sai_ids}
+        self.if_counters = \
+            {if_idx: self.db_conn[self.if_oid_namespace[if_idx]].get_all(mibs.COUNTERS_DB, mibs.counter_table(self.oid_sai_map[if_idx]))
+            for if_idx in self.oid_sai_map}
+
+        for oid in self.vlan_oid_sai_map:
+            if oid in self.vlan_oid_namespace:
+                db_index = self.vlan_oid_namespace[oid]
+                self.rif_counters[oid] = self.db_conn[db_index].get_all(mibs.COUNTERS_DB, mibs.counter_table(self.vlan_oid_sai_map[oid]), blocking=True)
+            
+        for db_index in self.port_rif_map:
+            for rif_sai_id in self.rif_port_map[db_index]:
+                port_sai_id = self.rif_port_map[db_index][rif_sai_id]
+                if port_sai_id in self.lag_sai_oid_map[db_index]:
+                    if_idx = self.lag_sai_oid_map[db_index][port_sai_id]
+                elif port_sai_id in self.sai_oid_map[db_index]:
+                    if_idx = self.sai_oid_map[db_index][port_sai_id]
+                else:
+                    continue
+                rif_counters = self.db_conn[db_index].get_all(mibs.COUNTERS_DB, mibs.counter_table(rif_sai_id), blocking=True)
```

Comment:
why blocking is True in this case ?

Reply:
The  previous code did a get_all with blocking=true, so kept the same. should not cause an issue because we get based on rif_sai_id from the specific db instance.

Reply:
but it will add to time taken ? same as original problem